components/provisioner/internal/director/directorclient.go

```diff
@@ -247,10 +248,34 @@ func (cc *directorClient) executeDirectorGraphQLCall(directorQuery string, tenan
 	req.Header.Set(AuthorizationHeader, fmt.Sprintf("Bearer %s", cc.token.AccessToken))
 	req.Header.Set(TenantHeader, tenant)
 
-	err := cc.gqlClient.Do(req, response)
-	if err != nil {
-		return apperrors.Internal("Failed to execute GraphQL query with Director: %s", err.Error())
+	if err := cc.gqlClient.Do(req, response); err != nil {
+		if egErr, ok := err.(gcli.ExtendedError); ok {
+			return mapDirectorErrorToProvisionerError(egErr).Append("Failed to execute GraphQL request to Director")
+		}
+		return apperrors.Internal("Failed to execute GraphQL request to Director: %v", err)
 	}
 
 	return nil
 }
+
+func mapDirectorErrorToProvisionerError(egErr gcli.ExtendedError) apperrors.AppError {
+	errorCodeValue, present := egErr.Extensions()["error_code"]
+	if !present {
+		return apperrors.Internal("Failed to read the error code from the error response. Original error: %v", egErr)
+	}
+	errorCode, ok := errorCodeValue.(float64)
```

Comment:
Questions: why casting the error code to `float64`?

Reply:
I checked the type of incoming error code from Director and it was `float64` even if it looks just like `21` for example. I tried to cast it to `int` anyway but it didn't work at all

Reply:
Thanks for insights. It's probably some GraphQL quirks.